arviz/plots/backends/matplotlib/plotpointinterval.py

```diff
@@ -0,0 +1,75 @@
+import numpy as np
+
+from ...plot_utils import calculate_point_estimate
+from ....stats import hdi
+
+
+def plot_point_interval(
```

Comment:
As Oriol had mentioned that we could use it for `plot_posterior`, not sure though. I think it should be better to add it to plot_utils

Reply:
I think we should already start to progressively move towards https://github.com/arviz-devs/arviz/wiki/Plot-hierarchy. This plot_point_interval could be the plot_atomic_box for example. Otherwise every plot needs to 

1) reimplement every single component in both backends (plot_posterior plots and labels the hdi interval, plot_violin has the exact same line/boxplot like formatting we have here yet we can't use that here, we need to reimplement it again for the dotplot; plot forest also does something very similar again completely reimplemented from scratch). 
2) when we make changes to facetting (with the skip dims PR for example or if we wanted to support what Tomas asked the other day on slack) we have to update every single plot, instead of updating the facet/grid level functions.